app/src/main/java/com/example/playlistmaker/find/FindActivity.kt

```diff
@@ -57,11 +132,96 @@ class FindActivity : AppCompatActivity() {
+                true
+            }
+            false
+        }
+        recyclerView = findViewById(R.id.findList)
         recyclerView.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
+        recyclerView.adapter = adapter
 
-        recyclerView.adapter = TrackAdapter(trackList)
+        updateButton.setOnClickListener {
+            onFindToInternet()
+        }
+
+    }
+
+    private fun clearTrackList(sharedPreferences: SharedPreferences) {
+        trackList.clear()
+        tracks.clear()
+        writeList(sharedPreferences)
+    }
+
+    fun loadList(sharedPreferences: SharedPreferences) { // функция выгрузки из sharedPreferences в переменную trackList
+        val tmpArray = read(sharedPreferences);
+        trackList.clear()
+        trackList.addAll(tmpArray)
+    }
+
+    fun writeList(sharedPreferences: SharedPreferences) { // функция сохранения в sharedPreferences из trackList
+        write(sharedPreferences, trackList)
+    }
+
+    private fun read(sharedPreferences: SharedPreferences): Array<Track> { // функция выгрузки из sharedPreferences служебная
+        val json = sharedPreferences.getString(TRACK_LIST_KEY, null) ?: return emptyArray()
+        return Gson().fromJson(json, Array<Track>::class.java)
+    }
+
+    private fun write(sharedPreferences: SharedPreferences, tracksList: List<Track>) { // функция сохранения в sharedPreferences служебная
+        val json = Gson().toJson(tracksList)
+        sharedPreferences.edit()
+            .putString(TRACK_LIST_KEY, json)
+            .apply()
+    }
+
+    private fun onFindToInternet() {
+        if (!input.isNullOrEmpty()) {
+            trackService.search(input.toString()).enqueue(object : Callback<TrackResponse> {
+                override fun onResponse(
+                    call: Call<TrackResponse>,
+                    response: Response<TrackResponse>
+                ) {
+                    if (response.code() == 200) {
+                        tracks.clear()
+                        if (response.body()?.results?.isNotEmpty() == true) {
+                            if (place_500.visibility == View.VISIBLE) {
+                                place_500.visibility = View.GONE
+                            }
+                            tracks.addAll(response.body()?.results!!)
+                            recyclerView.adapter?.notifyDataSetChanged()
+
+                        }
+                        if (tracks.isEmpty()) {
+                            Toast.makeText(
+                                applicationContext,
+                                R.string.what_go_wrong,
+                                Toast.LENGTH_LONG
+                            ).show()
+                            place_200.visibility = View.VISIBLE
+                        }
+                    } else {
+                        val code = response.code()
+                        val tmp = "" + R.string.error_of_server + " ${code}"
```

Comment:
```suggestion
                        val tmp = "" + getString(R.string.error_of_server) + " ${code}"
```
Здесь необходимо выводить строку, а не её айди